src/main/java/com/sprint/mission/discodeit/service/basic/BasicUserService.java

```diff
@@ -0,0 +1,156 @@
+    }
+
+    @Override
+    public List<UserDto> readAll() {
+        return userRepository.readAllContents()
+                .stream()
+                .map(this::toDto)
+                .toList();
+    }
+
+    @Override
+    public User updateUserField(UUID userId, UserUpdateRequest userUpdateRequest, BinaryContentCreateRequest optionalProfileCreateRequest) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new NoSuchElementException("User with id " + userId + " not found"));
+
+        String newUsername = userUpdateRequest.newUsername();
+        String newEmail = userUpdateRequest.newEmail();
+        final UUID nullableProfileId;
+        if (userRepository.existsByEmail(newEmail)) {
+            throw new IllegalArgumentException("User with email " + newEmail + " already exists");
+        }
+        if (userRepository.existsByUsername(newUsername)) {
+            throw new IllegalArgumentException("User with username " + newUsername + " already exists");
+        }
+
+        if (optionalProfileCreateRequest != null) {
+            nullableProfileId = binaryContentRepository.save(new BinaryContent(
+                    optionalProfileCreateRequest.fileName(),
+                    (long) optionalProfileCreateRequest.bytes().length,
+                    optionalProfileCreateRequest.contentType(),
+                    optionalProfileCreateRequest.bytes()
+            )).getId();
+        } else {
+            nullableProfileId = null;
+        }
+
+
+
+//        UUID nullableProfileId = optionalProfileCreateRequest
+//                .map(profileRequest -> {
+//                    Optional.ofNullable(user.getProfileId())
+//                            .ifPresent(binaryContentRepository::deleteById);
+//
+//                    String fileName = profileRequest.fileName();
+//                    String contentType = profileRequest.contentType();
+//                    byte[] bytes = profileRequest.bytes();
+//                    BinaryContent binaryContent = new BinaryContent(fileName, (long) bytes.length, contentType, bytes);
+//                    return binaryContentRepository.save(binaryContent).getId();
+//                })
+//                .orElse(null);
+
+        String newPassword = userUpdateRequest.newPassword();
+        user.update(newUsername, newEmail, newPassword, nullableProfileId);
+
+        return userRepository.save(user);
+    }
+
+    @Override
+    public void deleteUserById(UUID userId) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new NoSuchElementException(userId + " not found"));
+
+        Optional.ofNullable(user.getProfileId())
+                .ifPresent(binaryContentRepository::deleteById);
+        userStatusRepository.deleteById(userId);
+
+        userRepository.deleteById(userId);
+    }
+
+    private UserDto toDto(User user) {
+        Boolean online = userStatusRepository.findById(user.getId())
```

Comment:
프론트엔드 연동하는 테스트를 해보니 유저의 온라인 상태가 전부 오프라인으로 표시되는 오류가 있었는데요,  
아래처럼 코드를 수정해주면 버그가 해결될 것 같아요.
```suggestion
        Boolean online = userStatusRepository.findByUserId(user.getId())
```

![스크린샷 2025-03-16 오전 8 54 44](https://github.com/user-attachments/assets/a2f1a7f2-de23-49f9-851d-68677e3ca083)